rainmaker/egov-hrms/src/main/java/org/egov/hrms/service/MDMSService.java

```diff
@@ -147,7 +147,7 @@ public MdmsCriteriaReq prepareMDMSRequest(StringBuilder uri, RequestInfo request
 			for(String master: mapOfModulesAndMasters.get(module)) {
 				MasterDetail masterDetail=null;
 				if(module.equals(HRMSConstants.HRMS_AC_ROLES_MASTERS_CODE))
-					masterDetail = MasterDetail.builder().name(master).filter("*.code").build();
+					masterDetail = MasterDetail.builder().name(master).filter("[?(@.code != \"CITIZEN\")].code").build();
```

Comment:
Do move the filter expression up in the method or as a constant so that it can easily be refactored in the future.